src/Browscap/Writer/JsonWriter.php

```diff
@@ -1,34 +1,20 @@
 <?php
-/**
- * This file is part of the browscap package.
- *
- * Copyright (c) 1998-2017, Browser Capabilities Project
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
 declare(strict_types = 1);
 namespace Browscap\Writer;
 
 use Browscap\Data\DataCollection;
-use Browscap\Data\Expander;
+use Browscap\Data\Helper\TrimProperty;
 use Browscap\Filter\FilterInterface;
 use Browscap\Formatter\FormatterInterface;
 use Psr\Log\LoggerInterface;
 
 /**
- * Class JsonWriter
- *
- * @category   Browscap
- *
- * @author     Thomas Müller <mimmi20@live.de>
+ * this writer si responsible to create the browscap.json files
  */
-
-class JsonWriter implements WriterInterface, WriterNeedsExpanderInterface
+class JsonWriter implements WriterInterface
```

Comment:
These can all be `final` if they fully implement the `interface`

Reply:
changed